src/configuration.cc

```diff
@@ -124,6 +138,10 @@ int MetadataAgentConfiguration::ParseArguments(int ac, char** av) {
   boost::program_options::notify(flags);
 
   if (flags.count("help")) {
+    std::cout << flags_desc << std::endl;
+    return 0;
```

Comment:
Previously help returned 1, was this change intended?

Reply:
Ah, yes, it was. It's a bug fix. There's nothing wrong with asking the program to print a help message, so it shouldn't return a failure exit code.

Reply:
Turns out this bug fix introduced another bug -- we were using the nonzero return value of `ParseArguments` to cause the program to exit. Just fixed in another commit, which also added handling of argument parse errors and printing the version correctly.